recipes/luau/all/conanfile.py

```diff
@@ -17,18 +16,14 @@ class LuauConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://luau-lang.org/"
     topics = ("lua", "scripting", "typed", "embed")
-    package_type = "library"
+    package_type = "static-library"
```

Comment:
When making these changes, it's good if the PR message contains the rationale behind it to save some reviewing time :)

Since when is the library static-only? Was this an initial recipe bug or was support just dropped?

Reply:
@RubenRBS 
Sorry for the lack of information.
When I first wrote the recipe, I had created a patch to allow shared builds as well.
My goal was to make it versatile.

However, now that I think about it, this versatility is to be implemented for upstream, and at this time when upstream is restricting itself to static builds, I thought the conan recipe should also provide only static builds.